code/modules/mob/living/simple_animal/friendly/iriska.dm

```diff
@@ -0,0 +1,203 @@
+/mob/living/simple_animal/iriska
+	name = "Iriska"
+	desc = "The captain's own cat. Fat and lazy."
+	icon = 'icons/mob/iriska.dmi'
+	icon_state = "iriska"
+	health = 80
+	maxHealth = 80
+	wander = FALSE
+	canmove = FALSE
+	speak_emote = list("purrs.", "meows.")
+	emote_see = list("shakes her head.", "shivers.")
+	speak_chance = 0.75
+	meat_amount = 6
+	meat_type = /obj/item/reagent_containers/food/snacks/meat
+	response_help = "pets"
+	response_disarm = "rubs"
+	response_harm = "makes terrible mistake by kicking"
+	min_oxy = 16
+	minbodytemp = 223
+	maxbodytemp = 323
+	mob_size = MOB_HUGE
+	harm_intent_damage = 20
+	melee_damage_lower = 10
+	melee_damage_upper = 30
+	attacktext = "slashed"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	density = TRUE
+	bite_factor = 0.8
+	stomach_size_mult = 10
+
+	autoseek_food = FALSE // Original code for seeking food doesn't fit, Iriska will use slightly modified version
+	beg_for_food = FALSE
+	max_scan_interval = 10
+	eat_from_hand = FALSE
+
+var/atom/snack = null
+
+var/befriend_job = "Captain"
+
+var/list/mob/living/carbon/human/tolerated = list()
+var/list/mob/living/carbon/human/despised = list()
+
+/mob/living/simple_animal/iriska/fall_asleep()
+	return
+
+/mob/living/simple_animal/iriska/wake_up()
+	return
+
+/mob/living/simple_animal/iriska/Life()
+	.=..()
+
+	if(!stasis)
+
+		seek_food()
+		react_to_mob()
+
+/mob/living/simple_animal/iriska/proc/seek_food()
+	turns_since_scan++
+	if(turns_since_scan >= scan_interval)
+		turns_since_scan = 0
+		if(snack && (!(isturf(snack.loc) || ishuman(snack.loc)) || (foodtarget && !can_eat()) ))
+			snack = null
+			foodtarget = 0
+		if(!snack || !(snack.loc in oview(src, 4)))
+			snack = null
+			foodtarget = 0
+			if (can_eat())
+				for(var/obj/item/reagent_containers/food/snacks/S in oview(src,4))
+					if(S.name != "banana")
+						if(isturf(S.loc) || ishuman(S.loc))
+							snack = S
+							foodtarget = 1
+							break
+
+					if (!snack)
+						var/obj/item/reagent_containers/food/snacks/F = null
+						for(var/mob/living/carbon/human/H in oview(src, 4))
+							if(istype(H.l_hand, /obj/item/reagent_containers/food/snacks))
```

Comment:
Your logic here is flawed. If the user has something in both hands, the left hand is going to be ignored in favour of the right hand.
So, it could be the person is holding a good snack in their left hand, and a "banana" in their right hand, and Iriska will not recognize the good snack in the left hand because of the right hand banana.

Reply:
Checking active hand will be okay? Half the code are modified Ctrl+C Ctrl+V though, logic isn't mine.

Reply:
You can check active hand, yes